gizmos/search.py

```diff
@@ -3,50 +3,170 @@
 import sys
 
 from argparse import ArgumentParser
+from collections import defaultdict
 from .helpers import dict_factory
 
 
 def main():
     p = ArgumentParser()
     p.add_argument("db", help="SQLite database to search for labels")
     p.add_argument("text", nargs="?", help="Text to search")
+    p.add_argument(
+        "-L", "--label", help="Property for labels, default rdfs:label", default="rdfs:label"
+    )
+    p.add_argument("-S", "--short-label", help="Property for short labels, default is excluded")
+    p.add_argument(
+        "-s",
+        "--synonyms",
+        help="A property to include as synonym, default is excluded",
+        action="append",
+    )
     p.add_argument("-l", "--limit", help="Limit for number of results", type=int, default=30)
     args = p.parse_args()
-    sys.stdout.write(search(args.db, args.text, args.limit))
+    sys.stdout.write(
+        search(
+            args.db,
+            args.text,
+            label=args.label,
+            short_label=args.short_label,
+            synonyms=args.synonyms,
+            limit=args.limit,
+        )
+    )
 
 
-def search(db, text, limit=30):
-    names = get_names(db, text, limit)
+def search(db, text, label="rdfs:label", short_label=None, synonyms=None, limit=30):
+    names = get_names(db, text, limit, label=label, short_label=short_label, synonyms=synonyms)
     return json.dumps(names, indent=4)
 
 
-def get_names(db_path, text, limit):
+def get_names(db_path, text, limit, label="rdfs:label", short_label=None, synonyms=None):
     """Return a list of name details.
     Each item in the list is a dict containing 'display_name' (label) and 'value' (CURIE)."""
-    names = []
+    names = defaultdict(dict)
     with sqlite3.connect(db_path) as conn:
         conn.row_factory = dict_factory
         cur = conn.cursor()
         if text:
+            # Get labels
             cur.execute(
                 f"""SELECT DISTINCT subject, value
-                            FROM statements
-                            WHERE predicate = "rdfs:label"
-                            AND value LIKE "%{text}%"
-                            ORDER BY length(value)
-                            LIMIT {limit}"""
+                    FROM statements
+                    WHERE predicate = "{label}"
+                    AND value LIKE "%{text}%";"""
             )
+            for res in cur.fetchall():
+                term_id = res["subject"]
+                if term_id not in names:
+                    names[term_id] = dict()
+                names[term_id]["label"] = res["value"]
+
+            # Get short labels
+            if short_label:
```

Comment:
I would like the option for the short_label to be given as "ID" and then just return the subject. I think that needs a special case here.

Reply:
You can now do this with `--short-label ID`.